src/server/rest/v1/service/AuthorizationService.ts

```diff
@@ -43,6 +43,14 @@ export default class AuthorizationService {
         return entity.canCreate;
       case Action.DELETE:
         return entity.canDelete;
+      case Action.ASSIGN_CHARGING_STATIONS:
+        return entity['can' + Action.ASSIGN_CHARGING_STATIONS];
```

Comment:
I have created SiteAreaAuthorizationActions that inherits from AuthorizationActions but I did not know how to deal with the fact that this method is using the more generic type 'AuthorizationActions', so it does not know about "canAssignChargingStations" and the others. 
By downcasting, you mean using 'any' in this method instead of 'AuthorizationActions'?

Reply:
Use in the signature of the method:
`canPerfomAction(authActions: AuthorizationActions | SiteAreaAuthorizationActions, ...)`

In the code you downcast like this (by memory 😊):

```
case Action.ASSIGN_CHARGING_STATIONS:
  return (siteAreaAuthActions as SiteAreaAuthorizationActions).canAssignChargingStations
```

Reply:
Thank you!!!!